apps/maestro/src/ui/pages/InterchainTokenDetailsPage/TokenDetailsSection.tsx

```diff
@@ -18,20 +18,180 @@ import {
 } from "@axelarjs/ui";
 import { toast } from "@axelarjs/ui/toaster";
 import { maskAddress, Maybe } from "@axelarjs/utils";
-import { useMemo, useRef, useState, type FC } from "react";
+import {
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+  type FC,
+  type ReactNode,
+} from "react";
 import Identicon, { jsNumberForAddress } from "react-jazzicon";
 import Image from "next/image";
 
-import { createWalletClient, custom } from "viem";
+import { createWalletClient, custom, type Chain } from "viem";
 import { watchAsset } from "viem/actions";
 import { z } from "zod";
 
-import { STELLAR_CHAIN_ID, SUI_CHAIN_ID, useAccount } from "~/lib/hooks";
+import { useHederaTokenAssociation } from "~/features/hederaHooks";
+import {
+  HEDERA_CHAIN_ID,
+  STELLAR_CHAIN_ID,
+  SUI_CHAIN_ID,
+  useAccount,
+} from "~/lib/hooks";
 import { trpc } from "~/lib/trpc";
 import { hex64Literal } from "~/lib/utils/validation";
 import { ITSChainConfig } from "~/server/chainConfig";
 import { ChainIcon } from "~/ui/components/ChainsDropdown";
 
+type UseHederaAssociationArgs = {
+  isHederaChain: boolean;
+  tokenAddress: `0x${string}` | undefined;
+  connectedChain: Chain | undefined;
+  connectedAddress: string | undefined;
+  hasWallet: boolean;
+};
+
+function useHederaAssociation({
+  isHederaChain,
+  tokenAddress,
+  connectedChain,
+  connectedAddress,
+  hasWallet,
+}: UseHederaAssociationArgs) {
+  const {
+    checkHederaTokenAssociation,
+    associateHederaToken,
+    dissociateHederaToken,
+  } = useHederaTokenAssociation();
+
+  const [isAssociated, setIsAssociated] = useState<boolean | null>(null);
+  const [isCheckingAssociation, setIsCheckingAssociation] = useState(false);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  useEffect(() => {
+    let cancelled = false;
+    if (
+      !isHederaChain ||
+      !tokenAddress ||
+      !connectedChain ||
+      !connectedAddress ||
+      isSubmitting
+    )
+      return;
+
+    async function checkAssociation(address: `0x${string}`) {
+      setIsCheckingAssociation(true);
+      try {
+        const result = await checkHederaTokenAssociation(address);
+        if (!cancelled) setIsAssociated(Boolean(result));
+      } catch (e) {
+        console.error(e);
+        if (!cancelled) setIsAssociated(null);
+      } finally {
+        setIsCheckingAssociation(false);
+      }
+    }
+
+    void checkAssociation(tokenAddress);
+    return () => {
+      cancelled = true;
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [
+    connectedChain,
+    connectedAddress,
+    tokenAddress,
+    isHederaChain,
+    isSubmitting,
+  ]);
+
+  const onAssociate = async () => {
```

Comment:
Also, I noticed that this function doesn't call `setIsAssociated` to update the UI state when we press it, right? This is what I get on the FE.
If we use `useQuery` for the association check, this could invalidate the cache so it doesn't need to call `setIsAssociated` directly.